Gek62/hometask-4/my_hash.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+# Hash class extension created
+class MyHash
+  include Enumerable
+
+  OCCUPANCY_LEVEL = 0.7
+  DEFAULT_SIZE = 10
+
+  def initialize
+    @size = DEFAULT_SIZE
+    reset_dictionary
+  end
+
+  def []=(key, value)
+    resize! if full_dictionary?
+
+    slot = slot(key)
+    pair = hash_pair(slot, key)
+
+    add_hash(slot, key, value) if slot.empty? || pair.nil?
+
+    pair[1] = value if pair && !slot.empty?
+  end
+
+  def [](key)
+    collision = slot(key)
+
+    hash_pair(collision, key)&.last
+  end
+
+  def delete(key)
+    @pair_count -= 1
+
+    deleted_pair = slot(key).yield_self do |collision|
+      collision.dup - collision.delete_if { |k, _| k == key }
```

Comment:
I see 2 problems here. They're not critical at all, but anyway.
First of all, you don't have to use a placeholder argument (`_`) here, it's just redundant.
And moreover, `delete_if` method expects a block accepting _just one_ argument, so I can say it's redundant twice. :slightly_smiling_face: